gateway/oauth_manager.go

```diff
@@ -690,6 +691,15 @@ func (r *RedisOsinStorageInterface) SetClient(id string, client osin.Client, ign
 	log.Debug("Storing copy in set")
 
 	keyForSet := prefixClientset + prefixClient // Org ID
+
+	// In set, there is no option for update so the existing client should be removed before adding new one.
+	set, _ := r.store.GetSet(keyForSet)
+	for _, v := range set {
+		if strings.Contains(v, client.GetId()) {
+			r.store.RemoveFromSet(keyForSet, v)
```

Comment:
Do we really need this check? 
Feels like calling `r.store.RemoveFromSet(keyForSet, v)` every time, without getting fitst will do the same job, and will be even faster (because you do not need perform get)

Reply:
We need. There may be more than one client and user can update randomly.